src/pages/Swap/index.tsx

```diff
@@ -92,7 +92,40 @@ export function getIsValidSwapQuote(
   return !!swapInputError && !!trade && (tradeState === TradeState.VALID || tradeState === TradeState.SYNCING)
 }
 
-const formatAnalyticsEventProperties = (
+const formatApproveTokenTxnSubmittedEventProperties = (
+  approvalOptimizedTrade:
+    | Trade<Currency, Currency, TradeType>
+    | V2Trade<Currency, Currency, TradeType>
+    | V3Trade<Currency, Currency, TradeType>
+    | undefined
+) => {
+  if (!approvalOptimizedTrade) return {}
+  return {
+    chain_id: approvalOptimizedTrade.inputAmount.currency.chainId,
+    token_symbol: approvalOptimizedTrade.inputAmount.currency.symbol,
+    token_address: approvalOptimizedTrade.inputAmount.currency.isNative
+      ? NATIVE_CHAIN_ADDRESS
+      : approvalOptimizedTrade.inputAmount.currency.address,
+  }
+}
+
+const formatWrapTokenTxnSubmittedEventProperties = (
+  inputCurrency: Currency | null | undefined,
```

Comment:
why can it be `null` or `undefined`?

Reply:
currencies[Field.INPUT] and currencies[Field.OUTPUT] can both be null or undefined based on state of this react component. currencies is from the hook useDerivedSwapInfo().